test/tests/functional/pbs_resource_usage_log.py

```diff
@@ -0,0 +1,350 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2019 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+class TestResourceUsageLog(TestFunctional):
+
+    """
+    Test various scenarios in which resource usage is logged
+    in the accounting logs
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        attr1 = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, attr1)
+        attr2 = {'resources_available.mem': '200gb'}
+        self.server.manager(MGR_CMD_SET, NODE, attr2, id=self.mom.shortname)
+
+    def test_acclog_for_job_states(self):
+        """
+        Check accounting logs when a job completes successfully and when
+        a job is deleted in Q or R state
+        """
+        a = {'Resource_List.select': '1:ncpus=1:mem=200gb'}
+        j1 = Job(TEST_USER, a)
+        j1.create_eatcpu_job(40)
```

Comment:
I am just curious where is create_eatcpu_job( ) defined. Is it in  PTL library ?

Reply:
Thanks for pointing that. I had missed that commit in the PR. Added now